autogpts/autogpt/autogpt/agents/agent.py

```diff
@@ -64,157 +68,161 @@ class AgentConfiguration(BaseAgentConfiguration):
 
 class AgentSettings(BaseAgentSettings):
     config: AgentConfiguration = Field(default_factory=AgentConfiguration)
-    prompt_config: OneShotAgentPromptConfiguration = Field(
-        default_factory=(
-            lambda: OneShotAgentPromptStrategy.default_configuration.copy(deep=True)
-        )
-    )
 
 
-class Agent(
-    ContextMixin,
-    AgentFileManagerMixin,
-    WatchdogMixin,
-    BaseAgent,
-    Configurable[AgentSettings],
-):
-    """AutoGPT's primary Agent; uses one-shot prompting."""
-
+class Agent(BaseAgent, Configurable[AgentSettings]):
     default_settings: AgentSettings = AgentSettings(
         name="Agent",
-        description=__doc__,
+        description=__doc__ if __doc__ else "",
     )
 
-    prompt_strategy: OneShotAgentPromptStrategy
-
     def __init__(
         self,
         settings: AgentSettings,
         llm_provider: ChatModelProvider,
-        command_registry: CommandRegistry,
         file_storage: FileStorage,
         legacy_config: Config,
     ):
-        prompt_strategy = OneShotAgentPromptStrategy(
-            configuration=settings.prompt_config,
-            logger=logger,
+        super().__init__(settings, llm_provider)
+
+        # Components
+        self.system = SystemComponent(legacy_config, settings.ai_profile)
+        self.history = EventHistoryComponent(
+            settings.history,
+            self.send_token_limit,
+            lambda x: self.llm_provider.count_tokens(x, self.llm.name),
+            legacy_config,
+            llm_provider,
         )
-        super().__init__(
-            settings=settings,
-            llm_provider=llm_provider,
-            prompt_strategy=prompt_strategy,
-            command_registry=command_registry,
-            file_storage=file_storage,
-            legacy_config=legacy_config,
+        self.user_interaction = UserInteractionComponent(legacy_config)
+        self.file_manager = FileManagerComponent(settings, file_storage)
+        self.code_executor = CodeExecutorComponent(
+            self.file_manager.workspace,
+            settings,
+            legacy_config,
         )
+        self.git_ops = GitOperationsComponent(legacy_config)
+        self.image_gen = ImageGeneratorComponent(
+            self.file_manager.workspace, legacy_config
+        )
+        self.web_search = WebSearchComponent(legacy_config)
+        self.web_selenium = WebSeleniumComponent(legacy_config, llm_provider, self.llm)
+        self.context = ContextComponent(self.file_manager.workspace)
+        self.watchdog = WatchdogComponent(settings.config, settings.history)
+        self.prompt_strategy = OneShotAgentPromptStrategy(
+            OneShotAgentPromptStrategy.default_configuration, logger
+        )
```

Comment:
Need to set `use_functions_api` (previously implemented [here](https://github.com/Significant-Gravitas/AutoGPT/pull/7054/files#diff-989f831e9a5aa2e4e47738ecf38e238cb5f42842f5366df9fb480508cfc5d50fL106))